optuna/integration/_lightgbm_tuner/optimize.py

```diff
@@ -370,7 +384,8 @@ def __init__(
         self._parse_args(*args, **kwargs)
         self._start_time = None  # type: Optional[float]
         self._optuna_callbacks = optuna_callbacks
-        self._best_params = {}
+        self._best_booster_with_trial_number = None  # type: Optional[Tuple[lgb.Booster, int]]
```

Comment:
```suggestion
        self._best_booster_with_trial_number = None  # type: Optional[Tuple[Union[lgb.Booster, lgb.CVBooster], int]]
```

Reply:
Thank you so much for pointing out. I will make this change.

Reply:
Thank you for pointing out, please let me make this change.